torch/optim/lr_scheduler.py

```diff
@@ -1900,25 +1968,26 @@ def __init__(
         self.optimizer = optimizer
 
         # Validate total_steps
-        if total_steps is None and epochs is None and steps_per_epoch is None:
-            raise ValueError(
-                "You must define either total_steps OR (epochs AND steps_per_epoch)"
-            )
-        elif total_steps is not None:
+        if total_steps is not None:
             if total_steps <= 0 or not isinstance(total_steps, int):
                 raise ValueError(
                     f"Expected positive integer total_steps, but got {total_steps}"
                 )
             self.total_steps = total_steps
-        else:
+        elif epochs is not None and steps_per_epoch is not None:
```

Comment:
```suggestion
        elif epochs is not None or steps_per_epoch is not None:
```

Reply:
I think this condition should be `and` because both epochs and steps_per_epoch must have available value, otherwise it will fallback to the warning at line 1986. And maybe we should check epochos is an int before positive?

Reply:
@janeyx99, could you please review this again? I believe there may be some errors (as described above) in your suggested change.

Reply:
Ah yes, I was going off the DeMorgan law of not `epochs is None and steps_per_epoch is None` but you're right the code actually intends for both of these to be defined together.